### PR TITLE
charm_builder: install binary python packages (CRAFT-601)

### PR DESCRIPTION
**Pull request by cmatsuoka**
Charm plugin property `charm-binary-python-packages` can be used to
list binary packages to install from PyPI. Packages may still be
installed from sources if they're only available in source format.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -61,4 +61,4 @@ tomli==1.2.2
 typing-extensions==3.10.0.2
 urllib3==1.26.7
 zipp==3.6.0
-git+git://github.com/canonical/craft-cli.git@4af19f9c0da733321dc754be1180aea28f3feeb1
+git+https://github.com/canonical/craft-cli.git@4af19f9c0da733321dc754be1180aea28f3feeb1
```
Comment (facundobatista) [inline requirements-dev.txt:64]: Why this change?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -72,6 +72,7 @@ parts:
     plugin: nil
     build-packages:
       - python3-dev
+      - git
```
Comment (facundobatista) [inline snap/snapcraft.yaml:75]: What this is needed for?

**Review by facundobatista — CHANGES_REQUESTED** [review]
Added a couple of questions, thanks!

File: requirements-dev.txt (hunk 1 above)
Comment (cmatsuoka) [reply requirements-dev.txt:64]: Github is deprecating access using the unencrypted git protocol (see https://github.blog/2021-09-01-improving-git-protocol-security-github/). This is not effective yet but there was a brownout on Nov 02 when this PR was submitted. All entries using non-published packages will be removed before the next release.

File: snap/snapcraft.yaml (hunk 2 above)
Comment (cmatsuoka) [reply snap/snapcraft.yaml:75]: This was needed to be able to install packages using git+https and can be removed when those lines are removed. I will revert these changes since this brownout period ended.

**Review by facundobatista — APPROVED** [review]
Great, thanks!
